galsim/real.py

```diff
@@ -839,3 +905,426 @@ def _parse_files_dirs(file_name, image_dir, dir, noise_dir, sample):
+                            to use.  We strongly recommend leaving this parameter at its default
+                            value; see text above for details.  [default: galsim.Quintic()]
+    @param maxk             Optional maxk argument.  If you know you will be convolving the
+                            resulting `ChromaticRealGalaxy` with a "fat" PSF in a subsequent step,
+                            then it can be more efficient to limit the range of Fourier modes used
+                            when solving for the sum of separable profiles below.  [default: None]
+    @param pad_factor       Factor by which to internally oversample the Fourier-space images
+                            that represent the ChromaticRealGalaxy (equivalent to zero-padding the
+                            real-space profiles).  We strongly recommend leaving this parameter
+                            at its default value; see text in Realgalaxy docstring for details.
+                            [default: 4]
+    @param normalize_area   By default, the flux of the returned object is normalized such that
+                            drawing with exptime=1 and area=1 (the `drawImage` defaults) simulates
+                            an image with the appropriate number of counts for a 1 second HST
+                            exposure.  Setting this keyword to True will instead normalize the
+                            returned object such that to simulate a 1 second HST exposure, you must
+                            use drawImage with exptime=1 and area=45238.93416 (the HST collecting
+                            area in cm^2).  [default: False]
+    @param gsparams         An optional GSParams argument.  See the docstring for GSParams for
+                            details. [default: None]
+    @param logger           A logger object for output of progress statements if the user wants
+                            them.  [default: None]
+    Methods
+    -------
+
+    There are no additional methods for ChromaticRealGalaxy beyond the usual ChromaticObject
+    methods.
+    """
+    def __init__(self, real_galaxy_catalogs=None, SEDs=None, index=None, id=None, random=False,
+                 rng=None, k_interpolant=None, maxk=None, pad_factor=4., normalize_area=False,
+                 _imgs=None, _bands=None, _xis=None, _PSFs=None,  # hidden args
+                 gsparams=None, logger=None):
+        import numpy as np
+
+        if rng is None:
+            self.rng = galsim.BaseDeviate()
+        elif not isinstance(rng, galsim.BaseDeviate):
+            raise TypeError("The rng provided to ChromaticRealGalaxy constructor "
+                            "is not a BaseDeviate")
+        else:
+            self.rng = rng
+        self._rng = self.rng.duplicate()  # This is only needed if we want to make sure eval(repr)
+                                          # results in the same object.
```

Comment:
Not sure.  I was proposing to remove this line and see if anything breaks.  :)  So maybe try the same experiment with RealGalaxy.  I *think* we don't need it in either case, but I didn't verify.